DQM/BeamMonitor/plugins/AlcaBeamMonitor.cc

```diff
@@ -31,24 +29,22 @@ using namespace reco;
 
 //----------------------------------------------------------------------------------------------------------------------
 AlcaBeamMonitor::AlcaBeamMonitor(const ParameterSet& ps)
-    : parameters_(ps),
-      monitorName_(parameters_.getUntrackedParameter<string>("MonitorName", "YourSubsystemName")),
-      primaryVertexLabel_(
-          consumes<VertexCollection>(parameters_.getUntrackedParameter<InputTag>("PrimaryVertexLabel"))),
-      trackLabel_(consumes<reco::TrackCollection>(parameters_.getUntrackedParameter<InputTag>("TrackLabel"))),
-      scalerLabel_(consumes<BeamSpot>(parameters_.getUntrackedParameter<InputTag>("ScalerLabel"))),
-      beamSpotLabel_(parameters_.getUntrackedParameter<InputTag>("BeamSpotLabel")),
+    : monitorName_(ps.getUntrackedParameter<string>("MonitorName")),
```

Comment:
Could you explain why you have removed the second parameter `"YourSubsystemName"` here but not in `AlcaBeamMonitor::fillDescriptions`?

Reply:
Having it in both the `getUntrackedParameter` call and in `fillDescriptions` would be redundant. If the parameter is missing, the `fillDescriptions` call will inject the `"MonitorName"` string parameter into the module's PSet using the default `"YourSubsystemName"`. Therefore during the constructor call the parameter `"MonitorName"` is guaranteed to exist so no default value is needed for the `getUntrackedParameter`. So the change I made has exactly the same ultimate behavior as the original code.

In general, the framework group highly encourages everyone to stop specifying default values in `getUntrackedParameter` and instead use `fillDescriptions`.